JavaFXUserInterface/src/main/java/controllers/AssetsController.java

```diff
@@ -190,99 +213,162 @@ private void sortingSetUp() {
+     *
+     * @author Najim
+     */
+    private void searchAssets() {
+        searchedAssets = FXCollections.observableArrayList();
+        search.textProperty().addListener((observable, oldValue, newValue) -> {
+            searchedAssets.clear();
+            if (!newValue.equals("")) {
+                assets.stream().filter(o -> StringUtils.containsIgnoreCase(o.getRul().get(), newValue) || StringUtils.containsIgnoreCase(o.getName(), newValue)
+                        || StringUtils.containsIgnoreCase(o.getDescription(), newValue) || StringUtils.containsIgnoreCase(o.getSerialNo(), newValue)
+                        || StringUtils.containsIgnoreCase(o.getManufacturer(), newValue) || StringUtils.containsIgnoreCase(o.getCategory(), newValue)
+                        || StringUtils.containsIgnoreCase(o.getSite(), newValue) || StringUtils.containsIgnoreCase(o.getLocation(), newValue)
+                        || StringUtils.containsIgnoreCase(o.getRecommendation(), newValue) || StringUtils.containsIgnoreCase(assetTypeDAO.getNameFromID(o.getAssetTypeID()), newValue)).forEach(
+                        searchedAssets::add //Assets matched with search are added to the searchAssets list
+                );
+                if (searchedAssets.isEmpty()) {
+                    searchMatch = "No Match";
+                } else {
+                    searchMatch = "Match";
+                }
+            } else {
+                searchMatch = "No Search";
+            }
+
+            if (assetsTabPane.getSelectionModel().getSelectedItem().getId().equals("thumbnailTab")) {
+                assetsThumbPane.getChildren().clear();
+                generateThumbnails();
+            } else {
+                assetsListPane.getChildren().clear();
+                generateList();
+            }
+        });
+    }
+
+    /**
+     * Determines which list of assets to be displayed between the searchAssets list
+     * and the original assets list.
+     *
+     * @return assetsToDisplay
+     * @author Najim
+     */
+    public ObservableList<Asset> setAssetListToDisplay() {
+        ObservableList<Asset> assetsToDisplay;
+        if (!searchedAssets.isEmpty()) {
+            assetsToDisplay = FXCollections.observableArrayList(searchedAssets);
+        } else {
+            if (searchMatch.equals("No Match")) {
+                assetsThumbPane.getChildren().clear();
+                assetsToDisplay = null;
+            } else {
+                assetsToDisplay = FXCollections.observableArrayList(assets);
+            }
+        }
+        return assetsToDisplay;
+    }
+
     /**
      * Creates elements that are in the scene so the data can be displayed.
      *
      * @author Jeff
      */
     public void generateThumbnails() {
         ObservableList<Pane> boxes = FXCollections.observableArrayList();
+        ObservableList<Asset> assetsDisplayed = setAssetListToDisplay();
 
-        for (Asset asset : assets) {
+        if (assetsDisplayed != null) {
```

Comment:
I think it would be good to provide an alternative text to display on the page in the case where there are no assets to display. What I had in mind would be kind of a grey text appearing in the middle of the screen stating that there are "No results for your search" or something like that.

Reply:
"No results found" is now displayed if no match has been found from the search

Reply:
Well done. The only comment I would make is that it feels weird that the display of "No results found" is different for the list view than the thumbnail view. Was there a specific reason for it? Is it because of the flow pane? Otherwise, I would have left it the same as the list view.

Reply:
Yes, because of the flow pane I was not able to make it the same as the list view